src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataDictionary.cs

```diff
@@ -411,12 +411,50 @@ public static string FormatValue(object value, string format)
             {
                 return Convert.ToString(value, CultureInfo.CurrentCulture);
             }
+            else if (format == "{0:0000}-W{1:00}" && (value is DateTime || value is DateTimeOffset))
```

Comment:
This special case, if it must exist (absolutely must), should be conditional on something else. Don't compare format strings to constants.